widget/class-widget.php

```diff
@@ -61,6 +64,12 @@ public function widget( $args, $instance ) {
             )
          );
 
+         // Set default title if empty
+         $instance['title'] = empty( $instance['title'] ) ? 'Flexible Volunteer Opportunities' : $instance['title'];
+
+         // Get URL of page that [flexible_volunteer_opps] shortcode was used
+         $allOppsPageLink = get_permalink( $wpdb->get_results( 'SELECT ID FROM ' . $wpdb->base_prefix . 'posts WHERE post_content = "[flexible_volunteer_opps]" AND post_parent = 0')[0]->ID );
```

Comment:
This is a good start, but we'll need to make some tweaks. Here are some suggestions:

- [ ] For variables, we use all lowercase with underscores in between. Can we make the variable $all_opps_page_link?
- [ ] Take look at using get_var() instead of get_results() which should eliminate the need to grab the ID of the first array element.
- [ ] Right now the WHERE clause checks for the post_content = "[flexible_volunteer_opps]". The issue is that the content needs to equal exactly that and nothing else. If there is content above the shortcode or even a new paragraph it won't work. We need to use the LIKE clause here with % on either side of the string to make this more versatile.
- [ ] Make sure this works if you have the same shortcode on two different pages. Basically, pull the first result. 
- [ ] Right now we're not checking if the page is published, so even a draft's ID will show up. We need to make sure it's a published post.
- [ ] What made you decide to check post_parent as well? Feel free to write this as a reply.
- [ ] Before getting the permalink we need to see if any page has the shortcode on it. Right now if none does we get a PHP error. 
- [ ] How about turning the query into a method where we pass in the shortcode? That way we have one function that handles it for one-time and flexible opps. If you take that approach you likely need to use %s to pass in the variable.
 
Let me know if you'd like to discuss any of these.